### PR TITLE
Consolidate Junior and Water RC track code

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/ride/coaster/JuniorRollerCoaster.cpp [hunk 1]
```diff
@@ -1841,11 +1844,18 @@ static constexpr const uint32_t junior_rc_track_pieces_diag_blockbrakes[2][4] =
     },
 };
 
-void JuniorRCPaintTrackFlat(
-    PaintSession& session, const Ride& ride, uint8_t trackSequence, uint8_t direction, uint16_t height,
-    const TrackElement& trackElement, JuniorRCChainType chainType)
+template<JuniorRCSubType subType> constexpr uint8_t JuniorRCGetSubTypeOffset(const TrackElement& trackElement)
```
Comment (ZehMatt) [inline src/openrct2/ride/coaster/JuniorRollerCoaster.cpp:1847]: I think we use typically T prefix for template parameters, so TSubType.

Comment (Gymnasiast) [reply src/openrct2/ride/coaster/JuniorRollerCoaster.cpp:1847]: Is `TJuniorRCSubType` also okay? (It’s a parameter specific to the Junior/Water Coaster track.)

Comment (ZehMatt) [reply src/openrct2/ride/coaster/JuniorRollerCoaster.cpp:1847]: A bit long for my taste and its a strong type but I'm not against it.

File: src/openrct2/ride/coaster/JuniorRollerCoaster.cpp [hunk 2]
```diff
@@ -1844,12 +1844,12 @@ static constexpr const uint32_t junior_rc_track_pieces_diag_blockbrakes[2][4] =
     },
 };
 
-template<JuniorRCSubType subType> constexpr uint8_t JuniorRCGetSubTypeOffset(const TrackElement& trackElement)
+template<TJuniorRCSubType subType> constexpr uint8_t JuniorRCGetSubTypeOffset(const TrackElement& trackElement)
```
Comment (ZehMatt) [inline src/openrct2/ride/coaster/JuniorRollerCoaster.cpp:1847]: I think you misunderstood me, sorry if I didn't make it more clear, it should be `template<JuniorRCSubType TSubType>`, that is how we have it in some other places.

File: src/openrct2/ride/coaster/JuniorRollerCoaster.cpp [hunk 3]
```diff
@@ -1880,7 +1891,7 @@ void JuniorRCPaintStation(
         }
 
         // height += 2 (height)
-        if (trackElement.GetTrackType() == TrackElemType::EndStation && drawBlockBrake)
+        if (trackElement.GetTrackType() == TrackElemType::EndStation && TSubType == JuniorRCSubType::WaterCoaster)
```
Comment (Gymnasiast) [inline src/openrct2/ride/coaster/JuniorRollerCoaster.cpp:1894]:
```suggestion
        if (trackElement.GetTrackType() == TrackElemType::EndStation && TSubType == JuniorRCSubType::Junior)
```

File: src/openrct2/ride/coaster/JuniorRollerCoaster.cpp [hunk 4]
```diff
@@ -1900,7 +1911,7 @@ void JuniorRCPaintStation(
...
```
Comment (Gymnasiast) [inline src/openrct2/ride/coaster/JuniorRollerCoaster.cpp:1914]:
```suggestion
        if (trackElement.GetTrackType() == TrackElemType::EndStation && TSubType == JuniorRCSubType::Junior)
```

**Review by karst — APPROVED** [review]
After the fix that was just pushed I see no differences between this and develop.

**Review by ZehMatt — APPROVED** [review]
Looks good, always great to see code being removed and not added.

**Pull request merged**